src/Nikolaev/Task1/geometryengine.cpp

```diff
@@ -0,0 +1,152 @@
+
+    // Transfer index data to VBO 1
+    indexBuf.bind();
+    indexBuf.allocate(indices, 34 * sizeof(GLushort));
+}
+
+void GeometryEngine::setColor(QVector3D RGBcolors)
+{
+    std::array<GLfloat, 216u> vertices = {
+            // Vertex data for face 0
+            -1.0f, -1.0f, 1.0f, 0, 0, 1, RGBcolors[0], RGBcolors[1], RGBcolors[2],   // v0
+            1.0f, -1.0f, 1.0f, 0, 0, 1, RGBcolors[1], RGBcolors[1], RGBcolors[0],    // v1
+            -1.0f, 1.0f, 1.0f, 0, 0, 1, RGBcolors[2], RGBcolors[1], RGBcolors[2],    // v2
+            1.0f, 1.0f, 1.0f, 0, 0, 1, RGBcolors[0], RGBcolors[2], RGBcolors[2],     // v3
+
+            // Vertex data for face 1
+            1.0f, -1.0f, 1.0f, 1, 0, 0, RGBcolors[0], RGBcolors[1], RGBcolors[2],    // v4
+            1.0f, -1.0f, -1.0f, 1, 0, 0, RGBcolors[1], RGBcolors[1], RGBcolors[2],   // v5
+            1.0f, 1.0f, 1.0f, 1, 0, 0, RGBcolors[0], RGBcolors[1],  RGBcolors[2],    // v6
+            1.0f, 1.0f, -1.0f, 1, 0, 0, RGBcolors[2], RGBcolors[1], RGBcolors[2],    // v7
+
+            // Vertex data for face 2
+            1.0f, -1.0f, -1.0f, 0, 0, -1, RGBcolors[0], RGBcolors[1], RGBcolors[2],  // v8
+            -1.0f, -1.0f, -1.0f, 0, 0, -1, RGBcolors[0], RGBcolors[1], RGBcolors[0], // v9
+            1.0f, 1.0f, -1.0f, 0, 0, -1, RGBcolors[0], RGBcolors[2], RGBcolors[2],  // v10
+            -1.0f, 1.0f, -1.0f, 0, 0, -1, RGBcolors[0], RGBcolors[2], RGBcolors[2], // v11
+
+            // Vertex data for face 3
+            -1.0f, -1.0f, -1.0f, -1, 0, 0, RGBcolors[2], RGBcolors[1], RGBcolors[2], // v12
+            -1.0f, -1.0f, 1.0f, -1, 0, 0, RGBcolors[0], RGBcolors[2],  RGBcolors[2], // v13
+            -1.0f, 1.0f, -1.0f, -1, 0, 0, RGBcolors[0], RGBcolors[1], RGBcolors[1],  // v14
+            -1.0f, 1.0f, 1.0f, -1, 0, 0, RGBcolors[0], RGBcolors[1],  RGBcolors[1],  // v15
+
+            // Vertex data for face 4
+            -1.0f, -1.0f, -1.0f, 0, -1, 0, RGBcolors[0], RGBcolors[1], RGBcolors[2], // v16
+            1.0f, -1.0f, -1.0f, 0, -1, 0, RGBcolors[0], RGBcolors[1], RGBcolors[2],  // v17
+            -1.0f, -1.0f, 1.0f, 0, -1, 0, RGBcolors[1], RGBcolors[1], RGBcolors[2],  // v18
+            1.0f, -1.0f, 1.0f, 0, -1, 0, RGBcolors[0], RGBcolors[1], RGBcolors[2],   // v19
+
+            // Vertex data for face 5
+            -1.0f, 1.0f, 1.0f, 0, 1, 0, RGBcolors[0], RGBcolors[1], RGBcolors[0],    // v20
+            1.0f, 1.0f, 1.0f, 0, 1, 0, RGBcolors[0], RGBcolors[1], RGBcolors[1],     // v21
+            -1.0f, 1.0f, -1.0f, 0, 1, 0, RGBcolors[2], RGBcolors[1], RGBcolors[2],   // v22
+            1.0f, 1.0f, -1.0f, 0, 1, 0, RGBcolors[0], RGBcolors[1],  RGBcolors[2]    // v23
+    };
+
+    arrayBuf.bind();
+    arrayBuf.allocate(vertices.data(), vertices.size() * sizeof(GLfloat));
+}
+
+void GeometryEngine::drawCubeGeometry(QOpenGLShaderProgram *program)
+{
+    // Tell OpenGL which VBOs to use
+    arrayBuf.bind();
+    indexBuf.bind();
+
+    size_t stride = sizeof(GLfloat) * 9;
```

Comment:
```suggestion
    const auto stride = sizeof(GLfloat) * 9u;
```
Есть такое правило о экспертов с++, называется AAA - almost always auto. https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/